cocotb/types/__init__.py

```diff
@@ -0,0 +1,30 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+from .range import Range  # noqa: F401
+from .array import Array  # noqa: F401
+
+
+def concat(a: Array, b: Array) -> Array:
```

Comment:
To make things confusing, python already has `operator.concat`, and from the C API it is actually possible to make it mean something different from operator.add

Reply:
I was totally unaware that existed... But, if it doesn't work with it's own special methods in Python, that's a no go for us. I just wish Python had it's own concat operator :disappointed:

Reply:
Well, you could build a custom C type that connected those hooks to `__concat__` methods if you really wanted too, but that's probably not worth the trouble.